test/testsuites/opsrcdeletiontests.go

```diff
@@ -0,0 +1,96 @@
+package testsuites
+
+import (
+	"testing"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// OpSrcCreation is a test suite that ensures that the expected kubernetets resources are
+// created by marketplace after the creation of an OperatorSource.
+func OpSrcDeletion(t *testing.T) {
+	t.Run("operator-source-deletes-successfully", testOperatorSourceDeletesSuccessfully)
+}
+
+func testOperatorSourceDeletesSuccessfully(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	f := test.Global
+
+	// Get test namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	if err != nil {
+		t.Errorf("Could not get namespace: %v", err)
+	}
+
+	testOperatorSource := helpers.CreateOperatorSourceDefinition(namespace)
+
+	// Create the OperatorSource with no cleanup options.
+	err = helpers.CreateRuntimeObjectNoCleanup(f, ctx, testOperatorSource)
+	if err != nil {
+		t.Errorf("Could not create operator source: %v", err)
+	}
+
+	// Now that the OperatorSource has been created, let's poll until the
+	// Status.CurrentPhase is Succeeded
+	expectedPhase := "Succeeded"
```

Comment:
I am seeing this pattern being repeated. Please add a helper if possible.

Reply:
This can be done in #167 as well